Certificates beacon no longer check apiserver URL

alexandre-allard opened the pull request:
**Component**:
salt

**Context**:
In kubeconfig beacon we check for the apiserver URL with a specific hardcoded port (6443) which is not working for some of the kubeconfigs (e.g. calico), thus triggering a certificate renewal every time the beacon run for this certificate (luckily the certificate is not renewed because there is too much days left).

**Summary**:
Make the apiserver check optional in the `metalk8s_kubeconfig.validate()` function and no longer check it in kubeconfig beacon.

**Acceptance criteria**:
to be tested

alexandre-allard committed:
salt: Make kubeconfig apiserver check optional

Because we don't want to check this in certificate
beacon, we make it optional.

alexandre-allard committed:
salt: No longer check apiserver in kubeconfig beacon

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/cert-beacon-no-apiserver-check` into
`development/2.7`
* [w/2.8/bugfix/cert-beacon-no-apiserver-check](https://github.com/scality/metalk8s/compare/development/2.8...w/2.8/bugfix/cert-beacon-no-apiserver-check) will be merged into `development/2.8` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

TeddyAndrieux commented on salt/_modules/metalk8s_kubeconfig.py:
```diff
@@ -16,7 +16,7 @@ def __virtual__():
 
 def validate(filename,
              expected_ca_data,
```
Not related to this PR but to me, this one should be optional as well so that a user can easily check a kubeconfig by running `salt-call metalk8s_kubeconfig.validate <my-kubeconf>`

alexandre-allard committed:
salt: Make kubeconfig validate ca data optional

This way, the function can be called by a user
to check whether or not a kubeconfig needs to be
regenerated using the following command:

  salt-call metalk8s_kubeconfig.validate \
      /path/to/kubeconfig.conf

alexandre-allard commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/1b6d5af5a282c0888895951ea8b04ceedb82b980) did not succeed in branch w/2.8/bugfix/cert-beacon-no-apiserver-check.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
